src/Transform/ONNX/ONNXOpTransformPass.cpp

```diff
@@ -86,8 +86,12 @@ void ONNXOpTransformPass::runOnOperation() {
       dynamicPM.addNestedPass<func::FuncOp>(
           onnx_mlir::createShapeInferencePass());
     }
-    dynamicPM.addNestedPass<func::FuncOp>(
-        onnx_mlir::createConstPropONNXToONNXPass());
+    if ((/*enableConstantProp*/ onnx_mlir::OptimizationLevel >= 3 &&
+            !onnx_mlir::enableConstantProp) ||
```

Comment:
ONNXOpTransformPass has an unfortunate dependency on CompilerOptions because it reads the enableConvOptPass flag - it would be ideal if we could break that dependency, and not add another dependency on the new flags

Reply:
Either way I need the `OptimizationLevel`... that is the main reason why. So I still need to pull it from somewhere which is `src/Compiler/CompilerOptions.hpp`. This flag is contingent upon the level being specified at compile time.

Reply:
@hamptonm1 please remove direct references to `OptimizationLevel`.

You can simply add a parameter to the constructor
```
  ONNXOpTransformPass(int threshold, bool report, bool targetCPU,
      bool enableSimdDataLayoutOpt) {
```

so that that object only depend on the local variables.

I believe this is what @sorenlassen asked for.